_pytest/assertion/util.py

```diff
@@ -225,10 +225,18 @@ def _compare_eq_iterable(left, right, verbose=False):
     # dynamic import to speedup pytest
     import difflib
 
-    left = pprint.pformat(left).splitlines()
-    right = pprint.pformat(right).splitlines()
-    explanation = [u('Full diff:')]
-    explanation.extend(line.strip() for line in difflib.ndiff(left, right))
+    try:
+        left_formatting = pprint.pformat(left).splitlines()
+        right_formatting = pprint.pformat(right).splitlines()
+        explanation = [u('Full diff:')]
+    except RuntimeError:
```

Comment:
Why specifically `RuntimeError`? From the docs, `pformat` doesn't explicitly state that it may raise any specific exception... perhaps this should capture `Exception` instead? 

If that's the case, I think it would be best to put the `try/except` clause only over the `pprint.pformat` calls.